aws/modules/vpc/main.tf

```diff
@@ -0,0 +1,244 @@
+  })
+  depends_on = [
+    aws_internet_gateway.internet_gateway,
+    aws_eip.mod_nat_eip,
+    aws_subnet.public_subnet,
+  ]
+  lifecycle {
+    ignore_changes = [tags]
+  }
+}
+
+
+#-----------------------------------------------------------------------
+# Create AWS Internet gateway.
+resource "aws_internet_gateway" "internet_gateway" {
+  vpc_id = aws_vpc.vpc.id
+  tags = merge(
+    var.tags,
+    {
+      "Name" = format("%s-internet-gateway", aws_vpc.vpc.id)
+    }
+  )
+}
+
+#----------------------------------------------------------------------
+# create Elastic ip adress for the NAT
+resource "aws_eip" "mod_nat_eip" {
+  count = var.multi_az_nat_gateway * local.pri_az_count + var.single_nat_gateway * 1
+  tags = merge(var.tags, {
+    "Name" = format("%s-elasticIP-%s", var.name, local.pub_avilability_zones[count.index])
+  })
+  vpc = true
+}
+
+#-------------------------------------------------------------------------
+# Create AWS Security group 
+resource "aws_security_group" "security_group" {
+  name        = format("%s-security-group", var.name)
+  description = "Cluster communication with worker nodes"
+  vpc_id      = aws_vpc.vpc.id
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = merge(var.tags, {
+    "Name" = format("%s-security-group", var.name)
+    }
+  )
+}
+
+# Create AWS HTTPS Security group rule. Allow pods to communicate with the cluster API Server.
+resource "aws_security_group_rule" "https_security_group_rule" {
+  description              = "Allow pods to communicate with the cluster API Server"
+  from_port                = 443
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.security_group.id
+  source_security_group_id = aws_security_group.worker_security_group.id
+  to_port                  = 443
+  type                     = "ingress"
+}
+
+
+# Create AWS Workstation HTTPS Security group rule.
+resource "aws_security_group_rule" "workstation_https_group_rule" {
+  cidr_blocks       = [var.cluster_api_cidr]
+  description       = "Allow workstation to communicate with the cluster API Server"
+  from_port         = 443
+  protocol          = "tcp"
+  security_group_id = aws_security_group.security_group.id
+  to_port           = 443
+  type              = "ingress"
+}
+
+# Create AWS workers Security Group
+resource "aws_security_group" "worker_security_group" {
+  name        = format("%s-worker-security-group", var.name)
+  description = "Security group for all nodes in the cluster"
+  vpc_id      = aws_vpc.vpc.id
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = merge(var.tags, {
+    "Name"                                                   = format("%s-worker-security-group", var.name)
+    format("kubernetes.io/cluster/%s-eks-cluster", var.name) = "owned"
```

Comment:
Nitpick but put a new line after the first variable and the formatting will look better.